File: 98-ValidateBinarySearchTree/98-ValidateBinarySearchTree.cpp
// Last updated: 21/07/2026, 23:14:40
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

   bool BST(TreeNode* root,long &prev){

        if(!root) return 1;
        if(!BST(root->left,prev)){
            return 0;
        }
        if(root->val<=prev){
            return 0;
        }
        prev=root->val;
        return BST(root->right,prev);
   }
    bool isValidBST(TreeNode* root) {
       
       long prev=LONG_MIN;
       return BST(root,prev);
    }
};

File: 98-ValidateBinarySearchTree/98-ValidateBinarySearchTree.java
// Last updated: 21/07/2026, 23:14:33
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void inorder(TreeNode root, List<Integer> list) {
        if (root == null)
            return;

        inorder(root.left, list);
        list.add(root.val);
        inorder(root.right, list);

    }

    public boolean isValidBST(TreeNode root) {
        List<Integer> list = new LinkedList<>();
        inorder(root, list);
        for (int i = 1; i < list.size(); i++) {
            if (list.get(i) <= list.get(i - 1)) {
                return false;
            }
        }
        return true;

    }
}